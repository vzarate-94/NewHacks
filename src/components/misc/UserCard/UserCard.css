.user-card{
  display: flex;
  align-items: center;
  padding-left: 10px;
}

.user-card img{
  width: 40px;
  height: 40px;
  border: 1px solid #4d4d63;
  border-radius: 100%;
}

.user-info{
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin: 0;
  padding: 10px 10px 10px 15px;
}

.user-info * {
  margin: 0;
  color: rgb(215, 215, 215);
}

.user-info h4 {
  margin-bottom: 2px;
}

.user-info p {
  margin-top: 2px;
  font-size: 14px;
  color: rgb(145, 145, 145);
}