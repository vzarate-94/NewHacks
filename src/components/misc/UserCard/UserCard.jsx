import React from 'react'
import './UserCard.css'

const UserCard = (props) => {
    const { user } = props

    return (
        user &&
        <div className="user-card">
            <div className="user-info">
                <h4>{user.name}</h4>
                <h4>Posts</h4>
                
            </div>
        </div>
    )
}

export default UserCard