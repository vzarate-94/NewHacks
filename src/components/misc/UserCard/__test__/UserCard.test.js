import React from 'react';
import ReactDOM from 'react-dom';
import UserCard from '../UserCard';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'


afterEach(cleanup)
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<UserCard></UserCard>, div)
})

it('renders UserCard correctly', () => {
  const { getByTestId } = render(<UserCard user="Checo"/>);
  expect(getByTestId('component-test-1')).toHaveTextContent( `Posted By:`);
})