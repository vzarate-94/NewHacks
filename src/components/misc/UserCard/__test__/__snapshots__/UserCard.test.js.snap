// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Snapshot 1`] = `
<div
  data-testid="component-test-1"
>
  <div
    className="user-info"
  >
    <h4>
      Posted By: 
      <strong
        id="userName"
      />
    </h4>
  </div>
</div>
`;

exports[`matches Snapshot without user 1`] = `
<div
  data-testid="component-test-1"
/>
`;
